Handle wit files that have imports in a folder

Given a folder of Wit, some of the wit may [import other files](https://component-model.bytecodealliance.org/design/wit.html#worlds).

For example, wit in wasi-http has imports: https://github.com/WebAssembly/wasi-http/tree/main/wit

When you include this in the project you will get:


```
  Executing wit-bindgen...
EXEC : error : interface or world `imports` not found in package [C:\Users\jstur\projects\component\dotnet9-wasi-http-example\App.csproj]
       --> wit\deps\cli\command.wit:4:11
        |
      4 |   include imports;
        |           ^------
C:\Users\jstur\projects\component\dotnet9-wasi-http-example\.packages\bytecodealliance.componentize.dotnet.witbindgen\0.1.0-preview00003\buil 
d\ByteCodeAlliance.Componentize.DotNet.WitBindgen.targets(79,9): error MSB3073: The command "C:\Users\jstur\projects\component\dotnet9-wasi-h 
ttp-example\.packages\bytecodealliance.componentize.dotnet.witbindgen\0.1.0-preview00003\build\..\tools\win-x64\wit-bindgen.exe c-sharp wit\d 
eps\cli\command.wit  --runtime native-aot --out-dir obj\Debug\net8.0\wasi-wasm\wit_bindgen\" exited with code 1. [C:\Users\jstur\projects\com 
ponent\dotnet9-wasi-http-example\App.csproj]
```

This is because currently all files are executed individually  https://github.com/bytecodealliance/componentize-dotnet/blob/f33ea4382f57e60f39fa34ee27c543a14fbedefc/src/WitBindgen/build/ByteCodeAlliance.Componentize.DotNet.WitBindgen.props#L27

Ideally, we could execute the wit-bindgen command on a folder and only once.

This was found when trying out https://github.com/dicej/dotnet9-wasi-http-example.

See example in https://github.com/dicej/dotnet9-wasi-http-example/compare/main...jsturtevant:dotnet9-wasi-http-example:use-wasm-sdk?expand=1

--- a/README.md
+++ b/README.md
@@ -175,6 +175,17 @@ world root {
 
 This component can be used anywhere that WASI 0.2 components can be used. For example, use `wasm-tools compose` as illustrated above.
 
+### Referencing Wit Packages
+
+By default the project will find all wit files and execute wit-bindgen against each one. This is makes it easy to get started with a single wit file.  If you have more complicated wit files, then you can create wit packages. To use folder with all the wit in it you can add the following to your `.csproj`. 
+
+```xml
+<ItemGroup>
+    <Wit Remove="**\*.wit"  />
+    <Wit Include="wit-folder" World="wit-world" /> 
+</ItemGroup>
+```
+
 ### WIT strings and memory
 
 The calculator example above works easily because it doesn't need to allocate memory dynamically. Once you start working with strings, you must add an extra line to the `<PropertyGroup>` in your _host_ `.csproj` file (that is, the application that's _importing_ the interface):

--- a/test/WasmComponentSdkTest/WasmComponentSdkTest/SimpleProducerConsumerTest.cs
+++ b/test/WasmComponentSdkTest/WasmComponentSdkTest/SimpleProducerConsumerTest.cs
@@ -31,6 +31,13 @@ public class SimpleProducerConsumerTest
     }
 
     [Fact]
+    public void CanBuildComponentWithWitPackage()
+    {
+        var witInfo = GetWitInfo(FindModulePath($"../testapps/AppWithWitFolder/bin/{Config}", "appwithwitfolder-component.wasm"));
+        Assert.Contains("import test:pkg/folder", witInfo);
+    }
+
+    [Fact]
     public void CanComposeImportWithExport()
     {
         var composed = FindModulePath("../testapps/SimpleConsumer", "composed.wasm");

--- a/test/WasmComponentSdkTest/WasmComponentSdkTest/WasmComponentSdkTest.csproj
+++ b/test/WasmComponentSdkTest/WasmComponentSdkTest/WasmComponentSdkTest.csproj
@@ -28,6 +28,7 @@
 
     <ItemGroup>
         <ProjectReference Include="..\testapps\SimpleConsumer\SimpleConsumer.csproj" ReferenceOutputAssembly="false" />
+        <ProjectReference Include="..\testapps\AppWithWitFolder\AppWithWitFolder.csproj" ReferenceOutputAssembly="false"/>
     </ItemGroup>
 
 </Project>

--- a/test/WasmComponentSdkTest/testapps/AppWithWitFolder/AppWithWitFolder.csproj
+++ b/test/WasmComponentSdkTest/testapps/AppWithWitFolder/AppWithWitFolder.csproj
@@ -1,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <!-- Only needed when referencing the dependencies as projects. For package references, these are imported automatically. -->
+    <Import Project="..\..\..\..\src\WasmComponent.Sdk\ImportInDev.proj" />
+
+    <PropertyGroup>
+        <OutputType>Exe</OutputType>
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+
+        <InvariantGlobalization>true</InvariantGlobalization>
+        <RuntimeIdentifier>wasi-wasm</RuntimeIdentifier>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <ProjectReference Include="..\..\..\..\src\WasmComponent.Sdk\WasmComponent.Sdk.csproj" />
+    </ItemGroup>
+
+    <ItemGroup>
+        <Wit Remove="**\*.wit"  />
+        <Wit Include="wit" World="importstest" />   
+    </ItemGroup>
+</Project>

--- a/test/WasmComponentSdkTest/testapps/AppWithWitFolder/program.cs
+++ b/test/WasmComponentSdkTest/testapps/AppWithWitFolder/program.cs
@@ -1,0 +1,1 @@
+ImportstestWorld.wit.imports.test.pkg.v0_2_0.FolderInterop.Folder();

--- a/test/WasmComponentSdkTest/testapps/AppWithWitFolder/wit/imports.wit
+++ b/test/WasmComponentSdkTest/testapps/AppWithWitFolder/wit/imports.wit
@@ -1,0 +1,5 @@
+interface folder { folder: func();}
+
+world imports {
+    import folder;
+}

--- a/test/WasmComponentSdkTest/testapps/AppWithWitFolder/wit/world.wit
+++ b/test/WasmComponentSdkTest/testapps/AppWithWitFolder/wit/world.wit
@@ -1,0 +1,5 @@
+package test:pkg@0.2.0;
+
+world importstest {
+  include imports;
+}

